test/e2e/common/init_container.go

```diff
@@ -405,4 +409,363 @@ var _ = framework.KubeDescribe("InitContainer", func() {
 		Expect(len(endPod.Status.InitContainerStatuses)).To(Equal(2))
 		Expect(endPod.Status.ContainerStatuses[0].State.Waiting).ToNot(BeNil())
 	})
+
+	//**********************************************************************************************************************
+
+	It("should invoke init containers on a RestartNever pod", func() {
```

Comment:
rather than duplicating all the tests, can we factor this differently so we can run the test twice where we say whether or not to set the annotations as a boolean?